framework/src/problems/EigenProblem.C

```diff
@@ -143,6 +144,57 @@ EigenProblem::computeJacobianTag(const NumericVector<Number> & soln,
   _nl_eigen->disassociateMatrixFromTag(jacobian, tag);
 }
 
+void
+EigenProblem::computeMatricesTags(
+    const NumericVector<Number> & soln,
+    const std::vector<std::unique_ptr<SparseMatrix<Number>>> & jacobians,
+    const std::set<TagID> & tags)
+{
+  TIME_SECTION(_compute_jacobian_tag_timer);
+
+  if (jacobians.size() != tags.size())
+    mooseError("The number of matrices ",
+               jacobians.size(),
+               " does not equal the number of tags ",
+               tags.size());
+
+  _fe_matrix_tags.clear();
+
+  _nl_eigen->setSolution(soln);
+
+  _nl_eigen->disassociateAllTaggedMatrices();
+
+  unsigned int i = 0;
+  for (auto tag : tags)
+    _nl_eigen->associateMatrixToTag(*(jacobians[i++]), tag);
+
+  computeJacobianTags(tags);
+
+  i = 0;
+  for (auto tag : tags)
+    _nl_eigen->disassociateMatrixFromTag(*(jacobians[i++]), tag);
+}
+
+void
+EigenProblem::computeJacobianBlocks(std::vector<JacobianBlock *> & blocks)
+{
+  TIME_SECTION(_compute_jacobian_blocks_timer);
+
+  if (_displaced_problem)
+  {
+    _aux->compute(EXEC_PRE_DISPLACE);
+    _displaced_problem->updateMesh();
```

Comment:
Why are we calling `updateMesh`??

Reply:
I guess I copy these from somewhere.

Reply:
Hmm interesting. It seems expensive. I wonder why it would be necessary

Reply:
Possibly we could just delete it? If it broke anything, I will bring it back.

Reply:
`computeJacobianBlocks` is called by `PhysicsBasedPreconditioner`

Reply:
Apparently, I do not need it for my current toy examples. So I just deleted. If I need it in the future, I will bring it back, and then I will know the reason, LOL.

Reply:
Haha yes, I think it's good to avoid adding code we don't understand to new code